src/nti/site/subscribers.py

```diff
@@ -144,10 +162,11 @@ def threadSiteSubscriber(new_site, event):
                                             new_site_manager)
 
         setSite(new_fake_site)
-    else:
-        # Cancel traversal using a LocationError. This typically
-        # will get surfaced as a 404.
-        raise LocationError("Unknown kind of site", new_site, current_site)
+        return
+
+    # Cancel traversal using a LocationError. This typically
+    # will get surfaced as a 404.
+    raise LocationError("Unknown kind of site", new_site, current_site)
```

Comment:
This change has gotten me a bit further but I'm raising here now. When we're going to traverse in to one of our host sites, current_site is our IMainApplicationFolder and we fall all the way through to where we raise LocationError. Do we need a 

`(IMainApplicationFolder.providedBy(new_site) or IRootFolder.providedBy(new_site)) and IHostPolicyFolder.providedBy(new_site)` branch that does `setSite(new_site)`?

Reply:
I'm tempted to say that if we reach the end and have no more known special cases, we should just `setSite(new_site)` instead of raising. The rationale is thus:

1) We don't have unexpected 404s in production so we're not hitting this case. Therefore, it should have no production impact.
2) Raising is highly surprising.
3) We want to encourage more (correct, careful) use of scoped sites (e.g., courses can be sites, with their questions contained in their local site manager).
4) Adding more special cases devolves into a game of whack-a-mole. There's a general rule (when traversing into a site, apply it), lets use that. Because "Special cases aren't special enough to break the rules."

Reply:
That all sounds reasonable to me.